include/gazebo_lta_dynamics_plugin.h

```diff
@@ -0,0 +1,145 @@
+#include <gazebo/common/common.hh>
+#include <gazebo/common/Plugin.hh>
+#include "common.h"
+
+#include "Wind.pb.h"
+
+namespace gazebo {
+
+/// \brief Plugin class to implement the buoyancy and aerodynamics of a Lighter-Than-Air (LTA) airship.
+/// This plugin accepts the following SDF parameters:
+///
+/// <linkName>: Name of base link for receiving pose and and applying forces.
+/// <hullVolume>: The volume of the hull of the airship.
+/// <airDensity>: The density of air.
+/// <m11>: The added mass matrix term in row 1 column 1.
+/// <m22>: The added mass matrix term in row 2 column 2.
+/// <m26>: The added mass matrix term in row 2 column 6.
+/// <m33>: The added mass matrix term in row 3 column 3.
+/// <m35>: The added mass matrix term in row 3 column 5.
+/// <m44>: The added mass matrix term in row 4 column 4.
+/// <m53>: The added mass matrix term in row 5 column 3.
+/// <m55>: The added mass matrix term in row 5 column 5.
+/// <m62>: The added mass matrix term in row 6 column 2.
+/// <m66>: The added mass matrix term in row 6 column 6.
+/// <distCOV>: The distance from the nose to the Center of Volume (CoV).
+/// <distPotentialFlow>: The distance from the nose where the flow ceases to be potential.
+/// <distFinCenter>: The distance from the nose to the center of a fin.
+/// <distQuarterChord>: The distance from the center to the quarter chord length of a fin.
+/// <forceHullInviscidFlowCoeff>: The force coefficient of the inviscid flow contribution of the hull.
+/// <forceHullViscousFlowCoeff>: The force coefficient of the viscous flow contribution of the hull.
+/// <momentHullInviscidFlowCoeff>: The moment coefficient of the inviscid flow contribution of the hull.
+/// <momentHullViscousFlowCoeff>: The moment coefficient of the viscous flow contribution of the hull.
+/// <finNormalForceCoeff>: The fin normal force coefficient.
+/// <finStallAngle>: The fin stall angle [deg].
+/// <axialDragCoeff>: The axial drag coefficient.
+/// <windSubTopic>: The topic name to subscribe to wind velocity data.
+class GazeboLTADynamicsPlugin : public ModelPlugin {
+ public:
+  GazeboLTADynamicsPlugin()
+      : ModelPlugin() {}
+
+  virtual ~GazeboLTADynamicsPlugin();
+  virtual void Publish();
+ protected:
+  virtual void Load(physics::ModelPtr _model, sdf::ElementPtr _sdf);
+  virtual void OnUpdate(const common::UpdateInfo &);
+
+  virtual double Sign(double val);
+  virtual ignition::math::Vector3d LocalVelocity(ignition::math::Vector3d lin_vel, ignition::math::Vector3d ang_vel, ignition::math::Vector3d dist);
+  virtual double DynamicPressure(ignition::math::Vector3d vec);
+  virtual void UpdateForcesAndMoments();
+
+ private:
+  std::string namespace_;
+  std::string link_name_;
+
+  std::string wind_sub_topic_ = "/wind";
+  transport::SubscriberPtr wind_sub_;
+  typedef const boost::shared_ptr<const physics_msgs::msgs::Wind> WindPtr;
+
+  double param_hull_volume_;
+  double param_air_density_;
+  double param_m11_;
```

Comment:
Could you document what each parameter stands for?

Reply:
On lines 56-83 I documented what each of the gazebo plugin parameters are. I thought that would be sufficient. Would you rather that I restate that here for each variable as well? Let me know what you think and I'll make the necessary changes